components/clarity-lsp/src/common/state.rs

```diff
@@ -87,9 +94,23 @@ impl EditorState {
     }
 
     pub fn index_protocol(&mut self, manifest_location: FileLocation, protocol: ProtocolState) {
+        let base_location = manifest_location
+            .get_root_location_from_manifest_location(&manifest_location)
```

Comment:
Seems like an anti-pattern (self and manifest_location are the same) 🤔 
I initially made `get_root_location_from_manifest_location` to call it on `contract_location`s (which is still possible but not used anymore). wdyt?

Reply:
It looks like `get_root_location_from_manifest_location` doesn't need to be a method of `FileLocation` at all -- it doesn't use `self`. So what about just making it a function?